presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -114,12 +114,7 @@ public TaskResource(
         this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
         this.responseExecutor = requireNonNull(responseExecutor, "responseExecutor is null");
         this.timeoutExecutor = requireNonNull(timeoutExecutor, "timeoutExecutor is null");
-        if (communicationConfig.isBinaryTransportEnabled()) {
-            this.planFragmentCodec = planFragmentSmileCodec;
-        }
-        else {
-            this.planFragmentCodec = wrapJsonCodec(planFragmentJsonCodec);
-        }
+        this.planFragmentCodec = wrapJsonCodec(planFragmentJsonCodec);
```

Comment:
Maybe add a comment that plan fragment always use json due to flaky edge cases in smile as follow up?